docs/reference/extensions/go-framework.rst

```diff
@@ -0,0 +1,71 @@
+.. _go-framework-reference:
+
+go-framework
+----------------
+
+The Go extension streamlines the process of building Go application
+rocks.
+
+The extension builds and copies the Go binary file to the rock,
+inside the ``/app`` directory. By default, the base ``bare`` is used,
+to generate a lightweight image.
+
+
+.. note::
+    The Go extension is compatible with the ``bare`` and ``ubuntu@24.04``
+    bases.
+
+Project requirements
+====================
+
+To use the ``go-framework`` extension, there must be a ``go.mod`` file
+in the root directory of the project.
+
+
+``parts`` > ``go-framework/install-app`` > ``organize``
+=========================================================
+
+If the main package is in the base directory and the rockcraft name
+attribute is equal to the go module name, the name of the binary will
+be selected correctly, otherwise you can adjust it.
```

Comment:
>  otherwise you can adjust it.

Seems like I HAVE to change it with that part, otherwise it doesn't build.

<details>
<summary> go.mod </summary>

```go
module foo

go 1.23.0
```

</details>

<details>
<summary> rockcraft.yaml </summary>

```yaml
name: hello
version: '0.2' 
summary: TODO
description: TODO

base: bare 
build-base: ubuntu@24.04 
platforms:
    amd64:
    
extensions:
    - go-framework
```
</details>


<details>
<summary> Failure logs </summary>

```
Starting rockcraft, version 1.6.0
Logging execution to '/tmp/rockcraft.log'
*EXPERIMENTAL* extension 'go-framework' enabled
Extracting bare:latest
Extracted bare:latest
Initialising lifecycle
Installing build-packages
Installing build-snaps
Skipping pull for go-framework/base-layout (already ran)
Updating sources for go-framework/install-app (source changed)
Skipping pull for go-framework/runtime (already ran)
Skipping pull for pebble (already ran)
Reapplying go-framework/base-layout (previous layer changed)
Reapplying go-framework/install-app (previous layer changed)
Reapplying go-framework/runtime (previous layer changed)
Reapplying pebble (previous layer changed)
Skipping build for go-framework/base-layout (already ran)
Rebuilding go-framework/install-app ('organize' property changed)
:: + go mod download all
:: + go install -p 4 ./...
Skipping build for go-framework/runtime (already ran)
Skipping build for pebble (already ran)
Skipping stage for go-framework/base-layout (already ran)
Staging go-framework/install-app
/root/parts/go-framework/install-app/install/usr: No such file or directory
Failed to execute rockcraft in instance.
Full execution log: '/home/rough/.local/state/rockcraft/log/rockcraft-20241022-005012.610019.log'
```

</details>

---

But if I add the `parts.go-framework/install-app.organize` tweak, it works.

<details>
<summary> Changed rockcraft.yaml </summary>

```yaml
name: hello
version: '0.2' 
summary: TODO
description: TODO

base: bare 
build-base: ubuntu@24.04 
platforms:
    amd64:
    
extensions:
    - go-framework

parts:
  go-framework/install-app:
   organize:
     bin/foo: usr/local/bin/hello
```

</details>